### ISSUE TITLE
🐛 lint/correctness/noUnusedPrivateClassMembers not working as expected

### ISSUE THREAD

**Issue by ErnoW** [biomejs/biome#6165]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.13.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.15.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

`noUnusedPrivateClassMembers` behaves incorrectly on classes with multiple private methods, where some are used, and some are not used.

```json
{
  "linter": {
    "rules": {
      "correctness": {
        "noUnusedPrivateClassMembers": "error"
      }
    }
  }
}
```

Example correct behaviour of the linter (from https://github.com/biomejs/biome/blob/de989188284775939e2a88b7cb8ec0564788abb6/crates/biome_js_analyze/tests/specs/correctness/noUnusedPrivateClassMembers/invalid.ts)

```typescript
class TsBioo {
  private unusedProperty = 5;
  private unusedMethod() {}
}
```
```
> This private class member is defined but never used.
```

Example of incorrect behaviour. As soon as I add a private method/property that IS being used, then the previous errors are removed

```typescript
class TsBioo2 {
  private unusedProperty = 5;
  private unusedMethod() {}

  private usedProperty = 4
  public test(){
    return this.usedProperty;
  }
}
```
```
> (no error)
```

### Expected result

In the second example I would expect the linter to warn/error on `unusedProperty` and `unusedMethod`)
```typescript
class TsBioo2 {
  private unusedProperty = 5;
  private unusedMethod() {}

  private usedProperty = 4
  public test(){
    return this.usedProperty;
  }
}
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
